test/unit/test-performance.js

```diff
@@ -830,6 +830,86 @@ describes.realWin('PeformanceObserver metrics', {amp: true}, env => {
     }
   }
 
+  let fakeWin;
+  let windowEventListeners;
+  let performanceObserver;
+  let viewerVisibilityState;
+
+  function setupFakesForVisibilityStateManipulation() {
+    // Fake window to fake `document.visibilityState`.
+    fakeWin = {
+      Date: env.win.Date,
+      PerformanceObserver: env.sandbox.stub(),
+      addEventListener: env.sandbox.stub(),
+      removeEventListener: env.win.removeEventListener,
+      dispatchEvent: env.win.dispatchEvent,
+      document: {
+        addEventListener: env.sandbox.stub(),
+        hidden: false,
+        readyState: 'complete',
+        removeEventListener: env.sandbox.stub(),
+        visibilityState: 'visible',
+      },
+      location: env.win.location,
+      performance: {
+        getEntriesByType: env.sandbox.stub(),
+      },
+    };
```

Comment:
We have a `FakeWindow` fixture in fake-dom.js. You can instantiate it with a custom "describes" block:

```js
describes.fakeWin('my test name', /* spec */ {}, env => {
  it('test one', () => {
    env.win; // FakeWindow.
  });
});
```

Reply:
Resolving, see [this conversation](https://github.com/ampproject/amphtml/pull/24346#discussion_r320930137)